ext/drampower/src/BankStateVector.cc

```diff
@@ -0,0 +1,64 @@
+#include <vector>
+
+#include "BankStateVector.h"
+
+using namespace std;
+using namespace Data;
+
+BankStateVector::BankStateVector()
+{
+
+}
+
+size_t BankStateVector::size()
+{
+    return bank_state.size();
+}
+
+void BankStateVector::SetSize(int64_t size)
+{
+    bank_state.resize(static_cast<size_t>(size), BANK_PRECHARGED);
+    _num_active_banks = 0;
+}
+
+void BankStateVector::clear()
+{
+    _num_active_banks = 0;
+    bank_state.clear();
+}
+
+BankStateVector& BankStateVector::operator&= (BankStateVector& anotherBankVector)
+{
+    for (int i = 0; i < bank_state.size(); i++)
+    {
+        bank_state[i] = anotherBankVector.GetByIndex(i);
+    }
+    _num_active_banks = anotherBankVector.get_num_active_banks();
+    return *this;
+}
+
+void BankStateVector::SetByIndex(BankState newState, int index)
+{
+    if (bank_state[index] != newState)
+    {
```

Comment:
Our style guide says to put these curly braces on the same line as the if statement. Apologies for the annoying inconsistency.